crates/task-impls/src/quorum_proposal.rs

```diff
@@ -730,29 +621,19 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> QuorumProposalTaskState<TYPE
 
                 let view = certificate.view_number;
 
-                if self.quorum_membership.get_leader(view) == self.public_key {
-                    debug!(
-                        "Attempting to publish proposal after forming a View Sync Finalized Cert for view {}",
-                        *certificate.view_number
-                    );
-                    self.create_dependency_task_if_new(view, event_receiver, event_sender, event);
-                }
+                self.create_dependency_task_if_new(view, event_receiver, event_sender, event);
```

Comment:
Same here, shouldn't we create the dependency only if we are the leader?

Reply:
We do, but editing *each* call site was kind of a code smell, so I changed it so the function itself will just ignore non leader initiations.